src/Pelagos/Bundle/AppBundle/Controller/UI/DatasetDownloadReportController.php

```diff
@@ -200,4 +180,204 @@ private function formatSizeUnits($fileSizeBytes)
+                'UDI',
+                'TITLE',
+                'PRIMARY POINT OF CONTACT',
+                'PRIMARY POINT OF CONTACT EMAIL',
+                'TOTAL DOWNLOADS',
+                '# OF GOMRI DOWNLOADS',
+                '# OF GOOGLE DOWNLOADS',
+                'FILE SIZE(MB)'
+            ));
+        }
+
+        return $labels;
+    }
+
+    /**
+     * Get the correct query for the report.
+     *
+     * @param string     $reportName Name of the report.
+     * @param array|null $options    Additional parameters needed to run the query.
+     *
+     * @return Query
+     */
+    private function getQuery($reportName, array $options = null): Query
+    {
+        $container = $this->container;
+        $entityManager = $container->get('doctrine')->getManager();
+
+        //Query
+        if ($reportName === self::TIMESTAMP_REPORT) {
+            $qb = $entityManager->createQueryBuilder();
+            $query = $qb
+                ->select('log.creationTimeStamp, d.udi, log.payLoad')
+                ->from('\Pelagos\Entity\LogActionItem', 'log')
+                ->join('\Pelagos\Entity\Dataset', 'd')
+                ->where('log.subjectEntityId = d.id')
+                ->andWhere('log.subjectEntityName = ?1')
+                ->andWhere('log.actionName = ?2')
+                ->orderBy('log.creationTimeStamp', 'ASC')
+                ->setParameter(1, 'Pelagos\Entity\Dataset')
+                ->setParameter(2, 'File Download')
+                ->getQuery();
+        } else {
+            $queryString = 'SELECT dataset.udi,log.payLoad from ' .
+                LogActionItem::class . ' log join ' . Dataset::class . ' dataset with
+                log.subjectEntityId = dataset.id where log.actionName = :actionName and
+                log.subjectEntityName = :subjectEntityName and 
+                log.creationTimeStamp >= :startDate 
+                and log.creationTimeStamp <= :endDate order by dataset.udi ASC';
+
+            $query = $entityManager->createQuery($queryString);
+            $endDateInclusively = clone $options['endDate'];
+            $endDateInclusively = $endDateInclusively->add(new \DateInterval('P1D'));
+            $query->setParameters([
+                'actionName' => 'File Download',
+                'subjectEntityName' => 'Pelagos\Entity\Dataset',
+                'startDate' => $options['startDate'],
+                'endDate' => $endDateInclusively
+            ]);
+        }
+
+        return $query;
+    }
+
+    /**
+     * Get the file size for the dataset.
+     *
+     * @param Dataset $dataset The dataset instance.
+     *
+     * @return float
+     */
+    private function getFileSize(Dataset $dataset): float
+    {
+        $fileSize = 0;
+        $datasetSubmission = $dataset->getDatasetSubmission();
+        if ($datasetSubmission instanceof DatasetSubmission) {
+            $fileSize = $this->formatSizeUnits($datasetSubmission->getDatasetFileSize());
+        }
+
```

Comment:
If datasetFileSize is null, this will still actually return null despite initializing it to fileSize = 0 because the call to getDatasetFileSize will return null.